assets/blocks/course-outline/outline-block/existing-lessons-modal/lessons.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * External dependencies
+ */
+import { omitBy, uniq } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { CheckboxControl, Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Select lessons by filters.
+ *
+ * @param {Function} select  Data select function.
+ * @param {Object}   filters Filters object.
+ * @return {Object[]} Lessons.
+ */
+const selectLessons = ( select, filters ) => {
+	const foundLessons = select( 'core' ).getEntityRecords(
```

Comment:
I couldn't identify the reason, but I noticed that when searching for 1 char only, it's loading to search, but it's not filtering.

https://github.com/Automattic/sensei/assets/876340/ed289b0b-ec40-4ef6-9ed9-3c5d5dd17566

Reply:
Filtering happens on the backend, that's a standard functionality, brought by WP.
I noticed that even with two letters the results are still not very good.
One option I see here is to send request only when we have 3+ letters.
At the same time, I see a UI problem here: as we need also load it when it the search term is empty. I mean it might become confusing when it works and when doesn't.
So, I'd prefer to leave it as is, in this case.

Reply:
Could we use a similar implementation to how the existing questions modal works? We also need to fix the modal from jumping around. It should be a fixed height.

Reply:
> Could we use a similar implementation to how the existing questions modal works?

It does. Here is a link to questions (there is no special logic to handle the term or results): https://github.com/Automattic/sensei/blob/9e1d6ed78fafd4d2e4e94b51bfd935e3d66fa3d5/assets/blocks/quiz/quiz-block/questions-modal/questions.js#L41


>  We also need to fix the modal from jumping around. It should be a fixed height.

It was fixed earlier in https://github.com/Automattic/sensei/pull/7368/commits/9e1d6ed78fafd4d2e4e94b51bfd935e3d66fa3d5

Reply:
@renatho fixed here: https://github.com/Automattic/sensei/pull/7368/commits/46aea0702d5967aaa8fe166e5ff681581b8ae4c2

Reply:
Thank you for finding the solution and fixing it, @merkushin!

Reading it, I think it will impact any lesson query, no? If yes, maybe we should check somehow if it's the exact query we want?

Reply:
@renatho makes sense. I added another param to the request to distinguish our request.
Fixed here: https://github.com/Automattic/sensei/pull/7368/commits/2f90a553f1b6f94b01b4846629144db43b614263